src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11599,11 +11606,11 @@ void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix
             (attributes->get_vector_len() == AVX_512bit) ||
             (!_legacy_mode_vl) ||
             (attributes->is_legacy_mode())),"XMM register should be 0-15");
-    assert(((nds_enc < 16 && xreg_enc < 16) || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
+    assert(((nds_enc < 16 && xreg_enc < 16) || has_gpr || (!attributes->is_legacy_mode())),"XMM register should be 0-15");
   }
 
   clear_managed();
-  if (UseAVX > 2 && !attributes->is_legacy_mode())
+  if (UseAVX > 2 && !attributes->is_legacy_mode() || is_extended)
```

Comment:
This change then is not required.

Reply:
I think this change alone leaves the entrance to the evex encoding block too restrictive?  Since promoted instructions come in to this function with legacy set true, I think we would need to alter the legacy setting to make this work; that might be the best way to go. The assert change above asserts that we don't have both legacy mode and extended registers.  However the entrance to evex encoding, with the change, only admits !legacy mode.  With the change I observed cases where an extended gpr use case was vex encoded rather than evex encoded.

Reply:
As suggested, I changed the legacy attribute for vex instructions promoted to evex and this allows removing the "is_extended" term in the condition.  Thanks, done